ssl/statem/statem_clnt.c

```diff
@@ -2513,6 +2513,10 @@ static int tls_construct_cke_rsa(SSL *s, WPACKET *pkt, int *al)
         tmp_buf[0] = 0x70;
 # endif
 
+    /* Log the premaster secret, if logging is enabled. */
+    if (!ssl_log_rsa_client_key_exchange(s, encdata, enclen, pms, pmslen))
+        goto err;
+
```

Comment:
I'm thinking this should move down below the next condition, so it ends up just before this:

``` C
     s->s3->tmp.pms = pms;
```

That's to make sure it only gets logged if *everything* else worked out fine.